src/Orleans.Runtime/Catalog/ActivationData.cs

```diff
@@ -477,6 +479,11 @@ public void RecordRunning(Message message)
             // Note: This method is always called while holding lock on this activation, so no need for additional locks here
 
             numRunning++;
+            if (message.Direction != Message.Directions.OneWay && message.SendingGrain != null)
```

Comment:
Should we add a check for `SendingGrain.IsClient`? We cannot reasonably expect that two calls from the same client are in the same call chain. We could support call chains from the client, but we would need to have some other kind of identifier (eg, client messages could stick a unique id into the request context) - but that adds complexity.

So for now we should consider just avoiding adding clients here.

Reply:
Agree, check added.